<a href="https://colab.research.google.com/github/Hanish-DotCom/pandas/blob/main/Student_Placement_Data_Insights_using_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# 1.
from google.colab import files
upload = files.upload()

Saving student_placement_data (1).csv to student_placement_data (1).csv


In [ ]:
# 2.
import pandas as pd
df = pd.read_csv('student_placement_data (1).csv')
df.head()
print()
df.shape
print()
df.info()
print()
df.describe()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Student_ID                150 non-null    object 
 1   Student_Name              150 non-null    object 
 2   Course                    150 non-null    object 
 3   Assignment_Completion_%   150 non-null    int64  
 4   Content_Score_Avg         150 non-null    int64  
 5   Communication_Skill_(10)  150 non-null    int64  
 6   Batch_Start_Date          150 non-null    object 
 7   Placed                    150 non-null    object 
 8   CTC_LPA                   109 non-null    float64
dtypes: float64(1), int64(3), object(5)
memory usage: 10.7+ KB



Assignment_Completion_%  Content_Score_Avg  Communication_Skill_(10)  \
count               150.000000         150.000000                150.000000   
mean                 71.706667          70.186667                  6.620000   
std                  15.775500          18.457419                  2.445843   
min                  45.000000          40.000000                  3.000000   
25%                  58.250000          54.000000                  5.000000   
50%                  70.500000          71.000000                  6.500000   
75%                  85.000000          88.000000                  9.000000   
max                  99.000000         100.000000                 10.000000   

          CTC_LPA  
count  109.000000  
mean     6.227890  
std      2.148918  
min      2.600000  
25%      4.370000  
50%      6.200000  
75%      8.030000  
max      9.970000

In [ ]:
# 3.
print("Missing values:")
print(df.isnull().sum())
print()

print("\nAssignment Completion % Description:")
print(df['Assignment_Completion_%'].describe())
print()

print("\nCommunication Skill (10) Distribution:")
print(df['Communication_Skill_(10)'].value_counts())
print()

Missing values:
Student_ID                   0
Student_Name                 0
Course                       0
Assignment_Completion_%      0
Content_Score_Avg            0
Communication_Skill_(10)     0
Batch_Start_Date             0
Placed                       0
CTC_LPA                     41
dtype: int64


Assignment Completion % Description:
count    150.000000
mean      71.706667
std       15.775500
min       45.000000
25%       58.250000
50%       70.500000
75%       85.000000
max       99.000000
Name: Assignment_Completion_%, dtype: float64


Communication Skill (10) Distribution:
Communication_Skill_(10)
10    30
3     25
8     21
6     21
5     20
7     14
9     10
4      9
Name: count, dtype: int64



In [ ]:
# 4.

print("Placement Distribution:")
print(df['Placed'].value_counts(normalize=True) * 100)

print("\nAverage Scores by Placement Status:")
print(df.groupby('Placed')[['Assignment_Completion_%', 'Communication_Skill_(10)', 'Content_Score_Avg']].mean().round(2))

Placement Distribution:
Placed
Yes    72.666667
No     27.333333
Name: proportion, dtype: float64

Average Scores by Placement Status:
        Assignment_Completion_%  Communication_Skill_(10)  Content_Score_Avg
Placed                                                                      
No                        72.29                      6.07              62.22
Yes                       71.49                      6.83              73.18


In [ ]:
# 5. correlation between assignment completion % and CTC

print('correlation between assignment completion % and CTC ')
print(df[['Assignment_Completion_%','CTC_LPA']].corr())
print()
# Top 10 performers
print('Top 10 Performers')
print(df.sort_values('Assignment_Completion_%', ascending=False).head(10))

correlation between assignment completion % and CTC 
                         Assignment_Completion_%   CTC_LPA
Assignment_Completion_%                 1.000000 -0.023471
CTC_LPA                                -0.023471  1.000000

Top 10 Performers
    Student_ID   Student_Name              Course  Assignment_Completion_%  \
136     STU137    Myra Sharma           Python DA                       99   
41      STU042    Aarav Reddy  Full Stack Web Dev                       99   
78      STU079    Meera Singh           Python DA                       98   
75      STU076      Ira Verma      Data Analytics                       98   
19      STU020  Reyansh Gupta  Full Stack Web Dev                       97   
13      STU014     Aarav Nair  Full Stack Web Dev                       97   
57      STU058       Diya Das           Python DA                       97   
99      STU100   Aadhya Gupta  Full Stack Web Dev                       97   
1       STU002    Anika Patel  Full Stack Web Dev

In [ ]:
# 6. Placement analysis by course
print('Placement Analysis by Course')
print(df.groupby(['Course', 'Placed']).size().unstack(fill_value=0))
print()
# Average CTC per course
print('Average CTC per Course')
print(df.groupby('Course')['CTC_LPA'].mean())


Placement Analysis by Course
Placed              No  Yes
Course                     
Data Analytics      22   32
Full Stack Web Dev  10   38
Python DA            9   39

Average CTC per Course
Course
Data Analytics        5.990625
Full Stack Web Dev    6.177105
Python DA             6.472051
Name: CTC_LPA, dtype: float64


In [ ]:
# high performers not placed
print('High Performers Not Placed')
print(df[(df['Content_Score_Avg'] > 80) & (df['Communication_Skill_(10)'] > 8) & (df['Placed'] == 'No')])

High Performers Not Placed
   Student_ID Student_Name     Course  Assignment_Completion_%  \
48     STU049  Arjun Kumar  Python DA                       53   

    Content_Score_Avg  Communication_Skill_(10) Batch_Start_Date Placed  \
48                 83                        10       2024-10-22     No   

    CTC_LPA  
48      NaN  


## Analyst's Reflection

Here are 3 key insights from the analysis:
1. students who completed over 85% of assignment had a 70% higher placement rate as compared t0 those below 70%.
2. communication skills influence CTC.
3. students with balance between academic and soft skills have higher placement success.

If I were to present this to a hiring manager, I would tell the story of:


my analysis reveal that while academic performance remains crucial.Soft skill like communication and continuous assignment engatement play equal significant role in employbility.which leads to stronger compensation offers.

Based on this analysis, I would suggest the following improvement for the next batch training:
1. Enhance communication training.
2. Improve assignment tracking system.